api/stream_reactions.go

```diff
@@ -0,0 +1,359 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"slices"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/TUM-Dev/gocast/dao"
+	"github.com/TUM-Dev/gocast/model"
+	"github.com/TUM-Dev/gocast/tools"
+	"github.com/TUM-Dev/gocast/tools/realtime"
+	"github.com/getsentry/sentry-go"
+	"github.com/gin-gonic/gin"
+)
+
+type StreamReactionRoutes struct {
+	dao.DaoWrapper
+}
+
+// TODO: This can be modified to allow different reactions for different streams
+func (r StreamReactionRoutes) allowedReactions(c *gin.Context) {
+	c.JSON(http.StatusOK, tools.Cfg.AllowedReactions)
+}
+
+func (r StreamReactionRoutes) addReaction(c *gin.Context) {
+	cooldownSeconds := 10
+
+	tumLiveContext := c.MustGet("TUMLiveContext").(tools.TUMLiveContext)
+	user := tumLiveContext.User
+	stream := tumLiveContext.Stream
+
+	if stream == nil {
+		_ = c.Error(tools.RequestError{
+			Status:        http.StatusNotFound,
+			CustomMessage: "stream not found",
+		})
+		return
+	}
+
+	course, err := r.DaoWrapper.CoursesDao.GetCourseById(c, stream.CourseID)
+
+	if user == nil || err != nil {
+		_ = c.Error(tools.RequestError{
+			Status:        http.StatusInternalServerError,
+			CustomMessage: "user or course not found",
+		})
+		return
+	}
+
+	if !user.IsEligibleToWatchCourse(course) {
+		_ = c.Error(tools.RequestError{
+			Status:        http.StatusForbidden,
+			CustomMessage: "user not eligible to watch course",
+		})
+		return
+	}
```

Comment:
optional: these three conditionals seem like they could be a separate function that could be reused?